zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/UserTaskCompletingV2Applier.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.state.appliers;
+
+import io.camunda.zeebe.engine.state.TypedEventApplier;
+import io.camunda.zeebe.engine.state.immutable.UserTaskState.LifecycleState;
+import io.camunda.zeebe.engine.state.mutable.MutableProcessingState;
+import io.camunda.zeebe.engine.state.mutable.MutableUserTaskState;
+import io.camunda.zeebe.protocol.impl.record.value.usertask.UserTaskRecord;
+import io.camunda.zeebe.protocol.record.intent.UserTaskIntent;
+
+public final class UserTaskCompletingV2Applier
+    implements TypedEventApplier<UserTaskIntent, UserTaskRecord> {
+
+  private final MutableUserTaskState userTaskState;
+
+  public UserTaskCompletingV2Applier(final MutableProcessingState processingState) {
+    userTaskState = processingState.getUserTaskState();
+  }
+
+  @Override
+  public void applyState(final long key, final UserTaskRecord value) {
+    userTaskState.updateUserTaskLifecycleState(key, LifecycleState.COMPLETING);
+    userTaskState.createUserTaskIntermediateState(value);
```

Comment:
🔧 I wonder whether we can improve the understandability of this User Task Intermediate State somewhat by changing this method name. For example, `storeIntermediateState(value, LifecycleState.COMPLETING)` might help clarify that this is storing intermediate data for the completing state.

Reply:
Thanks for the suggestion, Nico!
I had a similar thought about linking the `user task record` with the `user task lifecycle`, which could also help with debugging. I've applied your suggestion, including the naming update, in this [commit](https://github.com/camunda/camunda/pull/22576/commits/65e06b31728360a30fcf5bc68a10bac29483f3fb#diff-495cacb2dfd86d4f86cddc70e7ce10b65ef46c2716d87dbbcc8e288f495a02d1).